src/radiation/radiation.cpp

```diff
@@ -409,16 +331,11 @@ void ablate::radiation::Radiation::RayInit() {
         }
     }
     /** Cleanup*/
-    VecRestoreArrayRead(cellGeomVec, &cellGeomArray) >> checkError;
-    RestoreRange(cellRange);
     DMDestroy(&radsearch) >> checkError;
 
     if (log) EndEvent();
 }
-
-PetscErrorCode ablate::radiation::Radiation::ComputeRHSFunction(PetscReal time, Vec solVec, Vec rhs) {
-    PetscFunctionBeginUser;
-
+const std::map<PetscInt, ablate::radiation::Radiation::Origin>& ablate::radiation::Radiation::Solve(Vec solVec, Vec rhs) {
```

Comment:
What does the radiation solver do with the solveVec and rhs?

Reply:
Hi, the absorptivity function needs the solution vector for mass fraction information. However, the solve no longer needs the rhs vector and I've removed this now.